css/prosemirror.scss

```diff
@@ -163,6 +163,18 @@ div.ProseMirror {
 		margin-bottom: 1em;
 	}
 
+	pre.frontmatter {
+		margin-bottom: 2em;
+		border-bottom: 2px solid var(--color-primary-element);
+	}
+
+	pre.frontmatter::before {
+		display: block;
+		content: 'Front Matter';
```

Comment:
Should be moved out of the styling as this would probably be bad for accessibility and is not translatable that way

Reply:
Ah, and it should be `Front matter` (Sentence case) as per https://docs.nextcloud.com/server/latest/developer_manual/design/foundations.html#wording

Reply:
That will make it a bit more complex, as this requires a NodeView.
For accessibility I do not see a problem as screen readers would announce that code block as 'front matter' which is fine.

But yes translation is a problem (even if I do not know if it should be translated, for example I think German users would be confused by *Titelei* instead of the technical term *front matter*).

I will try to find a good solution :)

Reply:
> That will make it a bit more complex, as this requires a NodeView.

I'm not sure... a [`renderHTML` function](https://tiptap.dev/guide/custom-extensions#render-html) might be enough.

```javascript
  renderHTML({ node, HTMLAttributes }) {
    return ['div', {},
      [ 'h3', {}, node.translatedHeading],
      this.parent?.(),
    ]
  }
```

I have not tested this. No idea if `this.parent` can be used in `renderHTML`. I took it from https://tiptap.dev/guide/custom-extensions#settings . If it does not work one could also repeat the `renderHTML` code from tiptaps code block there.

I'm also not sure if the translated heading needs to be in the node... You can probably also just translate it right in the `renderHTML` function.